packages/devtools-reps/src/reps/tests/long-string.js

```diff
@@ -0,0 +1,89 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const { shallow } = require("enzyme");
+const {
+  REPS,
+  getRep,
+} = require("../rep");
+let { LongStringRep } = REPS;
+const stubs = require("../stubs/long-string");
+
+function quoteNewlines(text) {
+  return text.split("\n").join("\\n");
+}
+
+describe("test Multiline / LongStringRep", () => {
+  it("selects LongString Rep", () => {
+    const stub = stubs.get("testMultiline");
+    expect(getRep(stub)).toEqual(LongStringRep.rep);
+  });
+
+  it("renders with expected text content for multiline string", () => {
+    const stub = stubs.get("testMultiline");
+    const renderedComponent = shallow(LongStringRep.rep({
+      object: stub
+    }));
+    expect(renderedComponent.text()).toEqual(quoteNewlines(`"${stub.initial}…"`));
+  });
+
+  it("renders with expected text content for multiline string with " +
+    "specified number of characters", () => {
+    const stub = stubs.get("testMultiline");
+
+    const renderedComponent = shallow(LongStringRep.rep({
+      object: stub,
+      cropLimit: 20
+    }));
+
+    expect(renderedComponent.text()).toEqual(`"a\\naaaaaaaaaaaaaaaaaa…"`);
+  });
+
+  it("renders with expected text content for multiline string when open", () => {
+    const stub = stubs.get("testMultiline");
+
+    const renderedComponent = shallow(LongStringRep.rep({
+      object: stub,
+      member: {open: true},
+      cropLimit: 20
+    }));
+
+    expect(renderedComponent.text())
+      .toEqual(quoteNewlines(`"${stub.initial}…"`));
+  });
+
+  it("renders with expected text content when grip has a fullText" +
+    "property and is open", () => {
+    const stub = stubs.get("testFullText");
+    const renderedComponent = shallow(LongStringRep.rep({
+      object: stub,
+      member: {open: true},
+      cropLimit: 20
+    }));
+
+    expect(renderedComponent.text()).toEqual(quoteNewlines(`"${stub.fullText}"`));
+  });
+
+  it("renders with expected text content when grip has a fullText " +
+    "property and is not open", () => {
+    const stub = stubs.get("testFullText");
+    const renderedComponent = shallow(LongStringRep.rep({
+      object: stub,
+      cropLimit: 20
+    }));
+
+    expect(renderedComponent.text()).toEqual(`"a\\naaaaaaaaaaaaaaaaaa…"`);
+  });
+
+  it("expected to omit quotes", () => {
+    const stub = stubs.get("testMultiline");
+    const renderedComponent = shallow(LongStringRep.rep({
+      object: stub,
+      cropLimit: 20,
+      useQuotes: false
+    }));
+
+    expect(renderedComponent.text()).toEqual(`a aaaaaaaaaaaaaaaaaa…`);
```

Comment:
Hey @jakub-c , thanks for working on this !
So the original assertion is : 
```js
is(renderedComponent.textContent, "a\naaaaaaaaaaaaaaaaaa…");
```

If you pay attention, the expected text uses double quotes (`"`), but you are using backtick quotes (`), and those handle line breaks differently. So in your case, when you have 
```js
  `a\na`
```
it does not mean there's a line break, but that the string actually contains the `\n` characters.
Switching to double quotes should resolve the issue I think.